sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubServiceAsyncClient.java

```diff
@@ -223,13 +218,8 @@
         if (requestOptions == null) {
             requestOptions = new RequestOptions();
         }
-        requestOptions.addHeader("Content-Type", contentType.toString());
-        requestOptions.addRequestCallback(httpRequest -> {
-            if (httpRequest.getHeaders().stream()
-                .noneMatch(header -> header.getName().equalsIgnoreCase("Content-Length"))) {
-                httpRequest.setHeader("Content-Length", String.valueOf(contentLength));
-            }
-        });
+        requestOptions.setHeader("Content-Type", contentType.toString());
+        requestOptions.setHeader("Content-Length", String.valueOf(contentLength));
```

Comment:
Make sure the WPS team are aware of these changes

Reply:
This is actually a change that needs to be know to LLC code generation.

cc: @weidongxu-microsoft

Reply:
@alzimmermsft
This method is actually manual code (the method is likely added by Srikanta as a easier-to-use convenient method for the LLC generated method, or another possibility is that this is generated by a very old codegen, and Srikanta keeps it for backward-compatibility when updating the Client), so there is no business with codegen here.

I see you already recorded the test, so this should be good.